_episodes/07-vector-data-in-python.md

```diff
@@ -149,126 +165,221 @@ array([109222.03325 , 469461.512625, 140295.122125, 510939.997875])
+> # A smaller bounding box in RD
+> xmin, xmax = (120_000, 135_000)
+> ymin, ymax = (485_000, 500_000)
+> ~~~
+> {: .language-python}
+> 3. Think of: what are the differences of the two methods? In what circumstances will you use `cx`? When will you use `clip_by_rect`?
+> {: .language-python}
+> > ## Answers
+> > 
+> > ~~~
+> > fields_cx = fields.cx[xmin:xmax, ymin:ymax]
+> > fields_rect = fields.clip_by_rect(xmin, ymin, xmax, ymax)
+> > fields_rect = fields_rect[~fields_rect.is_empty] # drop rows with empty geometries
+> > ~~~
+> > {: .language-python}
+> > 
+> > The method `cx` is an indexer, using square brackets, while `clip_by_rect` is a function using round brackets. The `cx` produces another `GeoDataFrame` with the attribute columns, while `clip_by_rect` produces a `GeoSeries`, with only the geometry information. `clip_by_rect` keep all entries, the entries outside the bounding box will be empty.
+> > 
+> > If attribute columns are needed, one should use `cx`. Otherwise, if only the geometries are needed, `clip_by_rect` can be used.
+> {: .solution}
+{: .challenge}
+
+
+## Export data to file
+
+From now on, we will continue with the cropped fields data `fields_cx`. We will save it to a shapefile (`.shp`) and use it later. The `to_file` function can be used for exportation:
+
+~~~
+fields_cx.to_file('data/fields_cropped.shp')
+~~~
+{: .language-python}
+
 
-![Extent image](../fig/E07-01-spatial_extent.png)
+This will write it to disk (in this case, in 'shapefile' format), containing only the data from our cropped area. It can be read again at a later time using the `read_file()` method we have been using above. Note that this actually writes multiple files to disk (`fields_cropped.cpg`, `fields_cropped.dbf`, `fields_cropped.prj`, `fields_cropped.shp`, `fields_cropped.shx`). All these files should ideally be present in order to re-read the dataset later, although only the `.shp`, `.shx`, and `.dbf` files are mandatory (see the [Introduction to Vector Data]({{site.baseurl}}/02-intro-to-vector-data) lesson for more information.)
 
-We can convert these coordinates to a bounding box or acquire the index of the dataframe to access the geometry. Either of these polygons can be used to clip rasters (more on that later).
 
 ## Selecting spatial features
-Sometimes, the loaded data can still be too large. We can cut it is to a even smaller extent using the `.cx` indexer (note the use of square brackets instead of round brackets, which are used instead with functions and methods):
+
+From now on, we will take in a point dataset `brogmwvolledigeset.zip`, which is the underground water monitoring wells. We will perform vector processing on this dataset, together with the cropped field polygons `fields_cropped.shp`.
+
+Let's read the two datasets.
 
 ~~~
-# Define a Boundingbox in RD
-xmin, xmax = (120_000, 135_000)
-ymin, ymax = (485_000, 500_000)
-cropfield_crop = cropfield.cx[xmin:xmax, ymin:ymax]
+fields =  gpd.read_file("data/fields_cropped.shp")
```

Comment:
Why re-reading the polygons?

Reply:
you are right, this comes from legacy, will remove.